key/keys.go

```diff
@@ -76,20 +80,28 @@ func (i *Identity) Equal(i2 *Identity) bool {
 }
 
 // SelfSign signs the public key with the key pair
-func (p *Pair) SelfSign() {
+func (p *Pair) SelfSign() error {
 	msg := p.Public.Hash()
-	signature, _ := AuthScheme.Sign(p.Key, msg)
+	signature, err := p.Public.Scheme.AuthScheme.Sign(p.Key, msg)
```

Comment:
and this is another reason why I had to add the Scheme to the Identity struct...